storage/remote/queue_manager_test.go

```diff
@@ -329,8 +327,9 @@ func TestCalculateDesiredsShards(t *testing.T) {
 		}
 		m.Start()
 		desiredShards := m.calculateDesiredShards()
+		shouldReshard := m.shouldReshard(desiredShards)
 		m.Stop()
-		if !c.reshard {
+		if !(c.reshard == shouldReshard) {
```

Comment:
I don't think we need to test the specific cases down below anymore as this test is targed at `shouldReshard`, one single `testutil.AssertEqual(t, c.reshard, shouldReshard)` should cover it.